### PR TITLE
Feat/714 post to put endpoint

### LINKED ISSUE #714: Modify #557 POST to PUT endpoint
The POST seedlot form endpoint developed in #557 can use some improvements as mentioned in the PR reviews.
As Ricardo is on Vacation, Craig will take over the implementation of these improvements. 
I made this separate ticket so I won't hijack Ricardo's 8-point work.

The main tasks in this issue are:
1. Update a database table so it can stores comments. See PR comment https://github.com/bcgov/nr-spar/pull/701#discussion_r1425364021
2. Delete all previously associated rows in different tables so all values can be updated. This way we can use this endpoint in the Seedlot edit and approval flow. Since this change will make it so that the endpoint either insert or update a record, the type classification of this endpoint will be changed from POST to PUT. See PR comment: https://github.com/bcgov/nr-spar/pull/701#discussion_r1427430811

### PR DESCRIPTION
**Pull request by craigyu**
# Description
Closes #714
There are a lot of files changed here, but most of them have been reviewed before. It might be easier to just review by checking out each commit pushed by me.

#### Changed
- store `intermOtherFacilityDesc` in column `interm_strg_locn` on table `seedlot`. Confirmed with @SLDonnelly  that this is ok.
- Updated to the post endpoint to a put. The endpoint now deletes all the dependant rows of seedlot then re-insert them again, so values can be updated when this endpoint is used for edition. 
- updated the usage of `deleteAllByIdInBatch` to `deleteAllById`, this gave me so much trouble, it kept giving me the error: `Row was updated or deleted by another transaction`. Turned out it's because `deleteAllByIdInBatch` "does not update the persistence context, requiring it to be cleared (the method bypasses the cache)". God bless this StackOverflow user! https://stackoverflow.com/a/64867259
- updated the usage of `saveAllAndFlush` to just `saveAll`, it works with save all so there's no reason to flush constantly.
- On front-end, added a max character counter to facility type description text input. Also changed the `onChange` to `onBlur` to reduce the # of re-rendering. 

### How was this tested?
- [ ] 🧠 Not needed
- [x] 👀 Eyeball
- [x] 🤖 Added tests

###  What gif/image best describes this PR or how it makes you feel?

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-722-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-722-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-722-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-spar-722-backend.apps.silver.devops.gov.bc.ca/)
[Frontend](https://nr-spar-722-frontend.apps.silver.devops.gov.bc.ca/)
[Oracle-API](https://nr-spar-722-oracle-api.apps.silver.devops.gov.bc.ca/)

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar/actions/workflows/merge-main.yml)

### CODE REVIEW

File: backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotOrchardService.java [hunk 1]
```diff
@@ -50,49 +50,22 @@ public void saveSeedlotFormStep4(Seedlot seedlot, SeedlotFormOrchardDto formStep
 
     if (!seedlotOrchards.isEmpty()) {
       log.info(
-          "{} previous records on the SeedlotOrchard table for seedlot number {}",
+          "deleting {} previous records on the SeedlotOrchard table for seedlot number {}",
```
Comment (mgaseta) [inline backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotOrchardService.java:53]: `deleting` without the capital `d`.

File: backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotService.java [hunk 2]
```diff
@@ -294,9 +293,13 @@ private void saveSeedlotFormStep3(Seedlot seedlot, SeedlotFormInterimDto formSte
...
     seedlot.setInterimStorageFacilityCode(formStep3.intermFacilityCode());
     // If the facility type is Other, then a description is required.
-    if (formStep3.intermFacilityCode() == "OTH") {
+    log.info("{} FACILITY TYPE CODE", formStep3.intermFacilityCode());
+    log.info("FACILITY TYPE Desc", formStep3.intermOtherFacilityDesc());
+    if (formStep3.intermFacilityCode().equals("OTH")) {
+      log.info("equal to OTH");
       if (formStep3.intermOtherFacilityDesc() == null
           || formStep3.intermOtherFacilityDesc().isEmpty()) {
+        log.info("hererererere");
```
Comment (mgaseta) [inline backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotService.java:302]: Remove this log.info

File: backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotService.java [hunk 3]
```diff
@@ -311,9 +314,11 @@ private void saveSeedlotFormStep5(
         "Saving Seedlot Form Step 5-Parent Tree SMP Mix for seedlot number {}", seedlot.getId());
 
     seedlotParentTreeService.saveSeedlotFormStep5(seedlot, seedlotFormParentTreeDtoList);
+    log.info("well well well");
```
Comment (mgaseta) [inline backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotService.java:317]: Remove this log too.

**Review by mgaseta — APPROVED** [review]
Nice work! :robot: :rocket: :100: 
This PUT endpoint will save us a lot of trouble when editing.
Added only a couple of comments to remove some logs.

File: backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotService.java (hunk 3 above)
Comment (craigyu) [reply backend/src/main/java/ca/bc/gov/backendstartapi/service/SeedlotService.java:317]: LOL, damn it

**Pull request merged**